test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs

```diff
@@ -570,17 +570,31 @@ public void DefaultContentRootIsApplicationBasePath()
         }
 
         [Fact]
-        public void DefaultApplicationNameWithNoStartupThrows()
+        public void DefaultWebHostBuilderWithNoStartupThrows()
         {
-            var builder = new ConfigurationBuilder();
             var host = new WebHostBuilder()
                 .UseServer(new TestServer());
 
-            var ex = Assert.Throws<ArgumentException>(() => host.Build());
+            var ex = Assert.Throws<InvalidOperationException>(() => host.Build());
+
+            Assert.Equal("No service for type 'Microsoft.AspNetCore.Hosting.IStartup' has been registered.", ex.Message);
```

Comment:
This is the exception thrown at https://github.com/aspnet/Hosting/blob/dev/src/Microsoft.AspNetCore.Hosting/Internal/WebHost.cs#L169. I think we might be able to do a little better by indicating that A startup need to be specified via UseStartup, Configure, or injecting IStartup.